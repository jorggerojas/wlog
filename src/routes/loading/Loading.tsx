import React from "react";
import svg from "../../load.svg";
import "../../styles/index.css";

interface LoadingProps {
  load: boolean;
}

const Loading = ({ load }: LoadingProps) => {
  return (
    <div
      className={`uk-width-1-1 ${
        load ? "uk-visible uk-animation-fade" : "uk-hidden uk-animation-fade"
      } uk-position-absolute uk-position-z-index uk-position-large uk-position-center uk-overlay uk-overlay-default uk-animation-fade`}
      style={{ background: "none", height: "70%" }}
    >
      <h3 className="uk-text-center uk-margin-large-bottom uk-text-bold">
        Loading, please wait...
      </h3>
      <div className="uk-text-center uk-margin">
        <img src={svg} alt="" style={{ position: "relative", width: "50%" }} />
      </div>
    </div>
  );
};

export default Loading;
