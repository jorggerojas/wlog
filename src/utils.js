import React from "react";
import { render, waitFor } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router, Route } from "react-router-dom";
import "@testing-library/jest-dom";

const renderWithRouter = (ui, { route = "/", ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  const finishLoading = () =>
    waitFor(() =>
      expect(utils.queryByText("Loading, please wait...")).toBeNull()
    );
  return {
    ...utils,
    finishLoading,
    history,
  };
};

export {
  Simulate,
  wait,
  render,
  cleanup,
  renderIntoDocument,
  fireEvent,
} from "@testing-library/react";
export { renderWithRouter };

export function renderWithProviders(
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
  apolloMocks
) {
  console.log("I am route inside renderWithProviders Wrapper:", route);
  return {
    ...render(
      <Router history={history}>
        <Route path={route}>{ui}</Route>
      </Router>
    ),
    history,
  };
}
