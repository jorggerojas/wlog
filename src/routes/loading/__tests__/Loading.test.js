import React from "react";
import { cleanup } from "@testing-library/react";
import Loading from "../Loading";
import { renderWithRouter } from "../../../utils";

afterEach(cleanup);

describe("<Loading/> component", () => {
  test('render with "true" props (visible)', async () => {
    const { container } = renderWithRouter(<Loading load={true} />);
    await expect(container.innerHTML.includes("uk-visible")).toBeTruthy();
    await expect(container).toHaveTextContent("Loading, please wait...");
    await expect(container).toMatchSnapshot();
  });
  test('render with "false" props (hidden)', async () => {
    const { container } = renderWithRouter(<Loading load={false} />);
    await expect(container.innerHTML.includes("uk-hidden")).toBeTruthy();
    await expect(container).toMatchSnapshot();
  });
});
