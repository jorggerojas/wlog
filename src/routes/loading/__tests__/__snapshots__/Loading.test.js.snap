// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading/> component render with "false" props (hidden) 1`] = `
<div>
  <div
    class="sc-bYEvPH cLJKVE uk-width-1-1 uk-hidden uk-animation-fade uk-margin-large-top uk-position-absolute uk-position-z-index uk-position-large
        uk-position-center uk-overlay uk-overlay-default uk-animation-fade"
  >
    <h3
      class="sc-pFZIQ hNoSbE uk-text-center uk-margin-large-top uk-text-bold"
    >
      Loading, please wait...
    </h3>
    <div
      class="uk-text-center uk-margin"
    >
      <img
        alt="Loading"
        class="sc-iqHYGH fKosLb"
        src="load.svg"
      />
    </div>
  </div>
</div>
`;

exports[`<Loading/> component render with "true" props (visible) 1`] = `
<div>
  <div
    class="sc-bYEvPH cLJKVE uk-width-1-1 uk-visible uk-animation-fade uk-margin-large-top uk-position-absolute uk-position-z-index uk-position-large
        uk-position-center uk-overlay uk-overlay-default uk-animation-fade"
  >
    <h3
      class="sc-pFZIQ hNoSbE uk-text-center uk-margin-large-top uk-text-bold"
    >
      Loading, please wait...
    </h3>
    <div
      class="uk-text-center uk-margin"
    >
      <img
        alt="Loading"
        class="sc-iqHYGH fKosLb"
        src="load.svg"
      />
    </div>
  </div>
</div>
`;
